source/features/download-folder-button.tsx

```diff
@@ -1,25 +1,44 @@
+import React from 'dom-chef';
+import {$, expectElement} from 'select-dom';
 import * as pageDetect from 'github-url-detection';
+import DownloadIcon from 'octicons-plain-react/Download';
 
 import features from '../feature-manager.js';
 import observe from '../helpers/selector-observer.js';
+import replaceElementTypeInPlace from '../helpers/recreate-element.js';
 
-function add({parentElement: deleteDirectoryItem}: HTMLAnchorElement): void {
-	const item = deleteDirectoryItem!.cloneNode(true);
-	const link = item.firstElementChild as HTMLAnchorElement;
+function add(menu: HTMLUListElement): void {
 	const downloadUrl = new URL('https://download-directory.github.io/');
 	downloadUrl.searchParams.set('url', location.href);
+
+	const item = menu.firstElementChild!.cloneNode(true);
+	item.role = 'none';
+	item.removeAttribute('tabindex');
+	item.removeAttribute('id');
+	item.removeAttribute('aria-keyshortcuts');
+	item.removeAttribute('aria-labelledby');
+
+	const link = item.firstElementChild instanceof HTMLAnchorElement
+		? item.firstElementChild
+		// Not a link on permalinks and archived repos
+		: replaceElementTypeInPlace(item.firstElementChild!, 'a');
 	link.href = downloadUrl.href;
-	link.textContent = 'Download directory';
-	link.removeAttribute('id');
-	link.removeAttribute('aria-keyshortcuts');
-	link.removeAttribute('aria-labelledby');
+	link.classList.add('no-underline', 'fgColor-inherit');
+	link.setAttribute('aria-keyshortcuts', 'c');
+
+	// Missing on permalinks and archived repos
+	$('svg', link)?.replaceWith(<DownloadIcon />);
```

Comment:
Since the link now appears at the top, I figured I'd also add an icon there:

<img width="307" alt="Screenshot" src="https://github.com/user-attachments/assets/f1215f2b-886e-4c76-93e8-74bb12225099">

However I'm now not really convinced that this should be at the top 🥲